File: integration-tests/src/test/java/dev/capylang/test/BitwiseTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class BitwiseTest {
    @Test
    void bitAnd() {
        assertThat(Bitwise.bitAnd(0b1100, 0b1010)).isEqualTo(0b1000);
    }

    @Test
    void bitNand() {
        assertThat(Bitwise.bitNand(0b1100, 0b1010)).isEqualTo(~0b1000);
    }

    @Test
    void bitOr() {
        assertThat(Bitwise.bitOr(0b1100, 0b1010)).isEqualTo(0b1110);
    }

    @Test
    void bitXor() {
        assertThat(Bitwise.bitXor(0b1100, 0b1010)).isEqualTo(0b0110);
    }

    @Test
    void bitNot() {
        assertThat(Bitwise.bitNot(0b1100)).isEqualTo(~0b1100);
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/Bitwise.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('Bitwise', () => {
    const bitwise = generatedModule('dev/capylang/test/Bitwise.js');
    assert.equal(bitwise.bitAnd(0b1100, 0b1010), 0b1000);
    assert.equal(bitwise.bitNand(0b1100, 0b1010), ~0b1000);
    assert.equal(bitwise.bitOr(0b1100, 0b1010), 0b1110);
    assert.equal(bitwise.bitXor(0b1100, 0b1010), 0b0110);
    assert.equal(bitwise.bitNot(0b1100), ~0b1100);
});
